### PR TITLE
[TESTING ONLY] Adds lots of ways to mess with batons for +DEV

### PR DESCRIPTION
**Pull request by JamieD1**
[TESTING ONLY] Adds lots of ways to mess with batons for +DEV

DO NOT MERGE

### CODE REVIEW

File: code/game/objects/items/stunbaton.dm [hunk 1]
```diff
@@ -34,6 +36,8 @@
 	var/preload_cell_type
 	///used for passive discharge
 	var/cell_last_used = 0
+	/// TESTING
+	var/obj/item/firing_pin/implant/mindshield/pin = /obj/item/firing_pin/implant/mindshield //standard firing pin for most guns
```
Comment (adamsong) [inline code/game/objects/items/stunbaton.dm]: This should probably default to an electronic pin, at the very least make the typepath of the pin var allow for it.

File: code/game/objects/items/stunbaton.dm [hunk 2]
```diff
@@ -143,6 +178,12 @@
 	add_fingerprint(user)
 
 /obj/item/melee/baton/attack(mob/M, mob/living/carbon/human/user)
+	/// TESTING
+	if(GLOB.batons_seconly)
```
Comment (adamsong) [inline code/game/objects/items/stunbaton.dm]:
```suggestion
	if(GLOB.batons_seconly && !GLOB.batons_normal)
```

File: code/game/objects/items/stunbaton.dm [hunk 3]
```diff
@@ -174,22 +215,32 @@
 
 	if(user.a_intent != INTENT_HARM)
 		if(status)
-			if(cooldown_check <= world.time)
+			/// TESTING
+			if(GLOB.batons_normal && GLOB.batons_cooldown)
```
Comment (adamsong) [inline code/game/objects/items/stunbaton.dm]:
```suggestion
			if(GLOB.batons_normal || GLOB.batons_cooldown)
```

File: code/game/objects/items/stunbaton.dm [hunk 4]
```diff
@@ -174,22 +215,32 @@
...
-				to_chat(user, span_danger("The baton is still charging!"))
+			///END TESTING
 		else
 			M.visible_message(span_warning("[user] has prodded [M] with [src]. Luckily it was off."), \
 							span_warning("[user] has prodded you with [src]. Luckily it was off."))
 	else
 		if(status)
-			if(cooldown_check <= world.time)
+			/// TESTING
+			if(GLOB.batons_normal && GLOB.batons_cooldown)
```
Comment (adamsong) [inline code/game/objects/items/stunbaton.dm]:
```suggestion
			if(GLOB.batons_normal || GLOB.batons_cooldown)
```

File: code/game/objects/items/stunbaton.dm [hunk 5]
```diff
@@ -283,3 +350,74 @@
...
+	set category = "Batons"
+	set name = "Toggle Batons Sec Only"
+	if(!check_rights(R_DEV))
+		return
+	GLOB.batons_seconly = !GLOB.batons_seconly
+	to_chat(usr, "Batons are now set to Sec only - TRUE/FALSE [GLOB.batons_seconly]")
+
+/datum/admins/proc/cmd_batons_cooldown()
+	set category = "Batons"
+	set name = "Toggle Batons Cooldown"
```
Comment (adamsong) [inline code/game/objects/items/stunbaton.dm:406]: Why is this just disable cooldowns? I thought we wanted to test different amounts of cooldown

**Review by adamsong — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stunbaton.dm (hunk 1 above)
Comment (JamieD1) [inline code/game/objects/items/stunbaton.dm]:
```suggestion
	var/obj/item/firing_pin/pin = /obj/item/firing_pin/implant/mindshield //mindshield implant
```

File: code/game/objects/items/stunbaton.dm [hunk 6]
```diff
@@ -283,3 +349,74 @@
...
+	/// TRUE IS 1 HIT
+	GLOB.batons_stam = !GLOB.batons_stam
+	to_chat(usr, "Batons are now set to 1 hit TRUE/FALSE - [GLOB.batons_stam]")
+
+/datum/admins/proc/cmd_batons_normal()
+	set category = "Batons"
+	set name = "Toggle Batons to Normal or Modified" ///Safety Net
+	if(!check_rights(R_DEV))
+		return
+	GLOB.batons_normal = !GLOB.batons_normal
```
Comment (JamieD1) [inline code/game/objects/items/stunbaton.dm]:
```suggestion
```

**Pull request closed without merging**